ktor-server/ktor-server-test-suites/jvm/src/io/ktor/server/testing/suites/SustainabilityTestSuite.kt

```diff
@@ -800,4 +805,34 @@ abstract class SustainabilityTestSuite<TEngine : ApplicationEngine, TConfigurati
             assertEquals(10000 * 13L, result.second)
         }
     }
+
+    @Test
+    fun testDoubleHost() {
+        createAndStartServer {
+            get("/") {
+                call.respond("OK")
+            }
+        }
+
+        socket {
+            val content = """
+                GET / HTTP/1.1
+                Host: www.example.com
+                Host: www.example2.com
+
+
+                """.trimIndent()
+
+            outputStream.bufferedWriter().apply {
+                write(content)
+                flush()
+            }
+
+            val response = inputStream.bufferedReader()
+            val status = response.readLine()
+
+            assertTrue(status.startsWith("HTTP/1.1 400"))
+            outputStream.close()
```

Comment:
I know that it doesn't matter in this case, but it's a bit painful to see that `inputStream` is not closed =)

Reply:
Socket is already closed at that moment, so `inputStream.close()` will fail with an exception